### ISSUE TITLE
amx_leave_message_enable not working

### ISSUE THREAD

**Issue by Fiercetree** [SuperCentral-Gaming-Community/AMX-Super-5.0.2#9]
I'm not getting any errors, just a function of the addon is not working. The amx_joinmessage_enable is working fine. But when someone leaves no message comes up.  
I do want to point out that these settings were enabled when I installed amx super plugin BUT I was using a third party join_leave plugin on my server. That third party plugin was working but after installing amx super, that third party plugin was pulling info from the amx super plugin. 

For example: the join_leave plugin would pull: %name% has joined!\nEnjoy the Server!\nCurrent Ranking is %rankpos%" In the font, color and position of the join_leave plugin's setting.  

So, I decided to use amx super's feature of this. 

STEPS I TOOK TO ATTEMPT TO RESOLVE THE ISSUE:
1. I disabled and removed the third party join_leave plugin from the server. The join feature on the amx super works but the leave option does not. 
2. Disabled all plugins except amx super and still no leave message. 
3. Inserted amx_joinmessage_enable 1 and amx_leave_message_enable 1 in server.cfg. Still no leave message. 
4. Changed cvar amx_leave_enable to 1 in console of the game, still no leave message.
5. Removed amx super from the server, downloaded and installed a fresh version. Still no leave message. 
6. Decided to just use third party plugin, downloaded and reinstalled join_leave and enabled it Disabled the join and leave feature on amx super, the third party plugin no longer works since removing it previously. 

The amx_joinmessage_enable is the only thing that has consistently worked. Is there another setting or another step I can take to enable the leave message?

**Comment by xLeviNx** [comment]
This has been a reoccurring issue. I’ll look into it when I get the chance

**Comment by Fiercetree** [comment]
Any update on this issue, just curious.

**Comment by xLeviNx** [comment]
Haven't been able to take a look at it, will review this week

**Comment by xLeviNx** [comment]
Can you test this
[amx_super_others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11299689/amx_super_others.zip)

**Comment by Fiercetree** [comment]
> Can you test this [amx_super_others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11299689/amx_super_others.zip)

Unfortunately, this file did not work at all. The amx super others menu is not even showing up in the plugin cvars list in-game and its features are not functioning, such as amx_joinmessage, etc. 

Note: After compiling this file for 1.82 and replacing the amxx and this sma file, I noticed these files are labeled as "amx_super_others" instead of "amx_super-others" like before. I'm not sure if this makes a difference, but I did make the necessary file name change in the third party plugin list in the plugin.ini

**Comment by xLeviNx** [comment]
> > Can you test this [amx_super_others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11299689/amx_super_others.zip)
> 
> Unfortunately, this file did not work at all. The amx super others menu is not even showing up in the plugin cvars list in-game and its features are not functioning, such as amx_joinmessage, etc.
> 
> Note: After compiling this file for 1.82 and replacing the amxx and this sma file, I noticed these files are labeled as "amx_super_others" instead of "amx_super-others" like before. I'm not sure if this makes a difference, but I did make the necessary file name change in the third party plugin list in the plugin.ini

Just to clarify, when changing the name, is it running? Can you see if the plugin is active?

**Comment by Fiercetree** [comment]
That's correct. It will not run.

 rcon amxx plugins
Currently loaded plugins:
       name                    version     author            file             status   
 [  1] Admin Base              1.8.2       AMXX Dev Team     admin.amxx       running  
 [  2] Admin Commands          1.8.2       AMXX Dev Team     admincmd.amxx    running  
 [  3] Admin Help              1.8.2       AMXX Dev Team     adminhelp.amxx   running  
 [  4] Slots Reservation       1.8.2       AMXX Dev Team     adminslots.amxx  running  
 [  5] Multi-Lingual System    1.8.2       AMXX Dev Team     multilingual.am  running  
 [  6] Menus Front-End         1.8.2       AMXX Dev Team     menufront.amxx   running  
 [  7] Commands Menu           1.8.2       AMXX Dev Team     cmdmenu.amxx     running  
 [  8] Players Menu            1.8.2       AMXX Dev Team     plmenu.amxx      running  
 [  9] Maps Menu               1.8.2       AMXX Dev Team     mapsmenu.amxx    running  
 [ 10] Plugin Menu             1.8.2       AMXX Dev Team     pluginmenu.amxx  running  
 [ 11] unknown                 unknown     unknown           antiflood.amxx   bad load 
 [ 12] Admin Chat              1.8.2       AMXX Dev Team     adminchat.amxx   running  
 [ 13] Scrolling Message       1.8.2       AMXX Dev Team     scrollmsg.amxx   running  
 [ 14] Info. Messages          1.8.2       AMXX Dev Team     imessage.amxx    running  

 [ 15] Admin Votes             1.8.2       AMXX Dev Team     adminvote.amxx   running  
 [ 16] NextMap                 1.8.2       AMXX Dev Team     nextmap.amxx     running  
 [ 17] Nextmap Chooser         1.8.2 f1    AMXX Dev Team     mapchooser.amxx  running  
 [ 18] TimeLeft                1.8.2       AMXX Dev Team     timeleft.amxx    running  
 [ 19] Pause Plugins           1.8.2       AMXX Dev Team     pausecfg.amxx    running  
 [ 20] Stats Configuration     1.8.2       AMXX Dev Team     statscfg.amxx    running  
 [ 21] StatsX                  1.8.2       AMXX Dev Team     statsx.amxx      running  
 [ 22] CS Misc. Stats          1.8.2       AMXX Dev Team     miscstats.amxx   running  
 [ 23] Quake Sounds            2.1         hleV              QuakeSounds.amx  running  
 [ 24] Ultimate Sounds         1.7         Dizzy / Bmann_42  ultimate_sounds  running  
 [ 25] Last man bets           0.9.4       JGHG              lastmanbets.amx  running  
 [ 26] Automatic Unstuck       1.5         NL)Ramon(NL       stuck.amxx       running  
 [ 27] Lame Connection Punish  1.2         MeRcyLeZZ         lame_connection  running  
 [ 28] AMXX Spy++              1.0         [DeathTV] Sid 6.  amx_spyplusplus  running  
 [ 29] Automatic knife duel    0.3         JGHG              automatic_knife  running  
 [ 30] Autoresponder/Advertis  0.5         MaximusBrood      ad_manager.amxx  running  
 [ 31] Ultimate Gore           1.6         JTP10181          amx_gore_ultima  running  
 [ 32] Pump Knockback (when s  1.0         v3x & Chronic     pump_knockback.  running  
 [ 33] AMX Super Cmd Manager   5.0.2       SuperCentral.co   amx_super-cmdma  running  
 [ 34] Amx Super Fun           5.0.2       SuperCentral.co   amx_super-fun.a  running  
 [ 35] AMX_Super Menu          5.0.2       SuperCentral.co   amx_super-menu.  running  
 [ 36] unknown                 unknown     unknown           amx_super_other  bad load 
 [ 37] Amx Super Serious       5.0.2       SuperCentral.co   amx_super-serio  running  
( 11) Load fails: Plugin file open error (plugin "antiflood.amxx")
( 36) Load fails: Load error 17 (invalid file format or version) (plugin "amx_super_others.amxx")

Bear in mind, I did not modify the file name of the compiled files and added them to the plugin folder and scripting folder. I simply modified the "**amx_super-others.amxx**" file name in my thirdparty plugin list in the plugin.ini to "**amx_super_other.amxx**"

It says Load 17 error, so I'm guessing it did not compile correctly?

*Using a web compiler, I got an error saying: 

"ROR: The "AMX Mod X 1.8.2" compiler process has been killed, because it has been running for more than 20 seconds.
Either your code is horribly broken, and made the compiler crash in an infinite loop,
or it's too complex to compile in such a short time.
Either way, this is done to spare CPU power on this webserver, so,
if you're certain your code should compile, do it locally."

So, I compiled it locally and got no errors:
//AMXXPC compile.exe
// by the AMX Mod X Dev Team


//// amx_super_others.sma
//
// Header size:           2356 bytes
// Code size:            19236 bytes
// Data size:            15052 bytes
// Stack/heap size:      16384 bytes
// Total requirements:   53028 bytes
// Done.
//
// Compilation Time: 0.27 sec

Same Load Error 17 though

**Comment by xLeviNx** [comment]
[amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11303501/amx_super-others.zip)

Try this one and compile

**Comment by Fiercetree** [comment]
> [amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11303501/amx_super-others.zip)
> 
> Try this one and compile

Was able to compile this successfully, amx_super-others is running. Still, the leave message is not working. Also, my join message is working but now it seems off. It says:

BEWARE\                                         
Wolf has joined.

instead of

BEWARE!
Wolf has joined.

Nothing was changed in amx_super config:
// Default value: "BEWARE!\n%name% has joined."
amx_join_message "BEWARE!\n%name% has joined."

**Comment by xLeviNx** [comment]
Can you show me what your leave message says?

**Comment by xLeviNx** [comment]
[amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11304514/amx_super-others.zip)

Try this one

**Comment by Fiercetree** [comment]
> Can you show me what your leave message says?

// Default value: 1
amx_leave_message_enable 1

// Message to show when a player leaves the server.
// Note: %name% & %rankpos% will be replaced with their appropriate values.
// Note: Not used when amx_leavemessage_enable is 0.
// 
// Default value: "%name% has left!\nCome back soon."
amx_leavemessage "%name% has left!\nCome back soon."

**Comment by Fiercetree** [comment]
> [amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11304514/amx_super-others.zip)
> 
> Try this one

Same result
Still, the leave message is not working. Also, my join message is working but it is still off, like before

**Comment by xLeviNx** [comment]
Are you running the new plugin? Just tested and the default message works along with anything else that can be added, but the leave msg should be different

**Comment by xLeviNx** [comment]
[amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11304630/amx_super-others.zip)

This one should work. After uploading and testing, can you provide the following info:

meta list
amxx plugins
version
status

**Comment by Fiercetree** [comment]
> [amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11304630/amx_super-others.zip)
> 
> This one should work. After uploading and testing, can you provide the following info:
> 
> meta list amxx plugins version status

This is info from my server using the last amx_super-others file you posted. Again, I am still getting the join message that is still off and the leave message still does not work.

] meta list
Currently running plugins:
 [  1] AMX Mod X, v1.8.2, Feb 14 2013, by AMX Mod X Dev Team, see http://www.amxmodx.org
 [  2] YaPB, v4.3.729, Jan 28 2023, by YaPB Project, see https://yapb.jeefo.net/
 [  3] Fun, v1.8.2, Feb 14 2013, by AMX Mod X Dev Team, see http://www.amxmodx.org
 [  4] Engine, v1.8.2, Feb 14 2013, by AMX Mod X Dev Team, see http://www.amxmodx.org
 [  5] FakeMeta, v1.8.2, Feb 14 2013, by AMX Mod X Dev Team, see http://www.amxmodx.org
 [  6] CStrike, v1.8.2, Feb 14 2013, by AMX Mod X Dev Team, see http://www.amxmodx.org
 [  7] CSX, v1.8.2, Feb 14 2013, by AMX Mod X Dev Team, see http://www.amxmodx.org/
 [  8] Ham Sandwich, v1.8.2, Feb 14 2013, by AMX Mod X Dev Team, see http://www.amxmodx.org
8 plugins


Currently loaded plugins:
       name                    version     author            file             status   
 [  1] Admin Base              1.8.2       AMXX Dev Team     admin.amxx       running  
 [  2] Admin Commands          1.8.2       AMXX Dev Team     admincmd.amxx    running  
 [  3] Admin Help              1.8.2       AMXX Dev Team     adminhelp.amxx   running  
 [  4] Slots Reservation       1.8.2       AMXX Dev Team     adminslots.amxx  running  
 [  5] Multi-Lingual System    1.8.2       AMXX Dev Team     multilingual.am  running  
 [  6] Menus Front-End         1.8.2       AMXX Dev Team     menufront.amxx   running  
 [  7] Commands Menu           1.8.2       AMXX Dev Team     cmdmenu.amxx     running  
 [  8] Players Menu            1.8.2       AMXX Dev Team     plmenu.amxx      running  
 [  9] Maps Menu               1.8.2       AMXX Dev Team     mapsmenu.amxx    running  
 [ 10] Plugin Menu             1.8.2       AMXX Dev Team     pluginmenu.amxx  running  
 [ 11] unknown                 unknown     unknown           antiflood.amxx   bad load 
 [ 12] Admin Chat              1.8.2       AMXX Dev Team     adminchat.amxx   running  
 [ 13] Scrolling Message       1.8.2       AMXX Dev Team     scrollmsg.amxx   running  
 [ 14] Info. Messages          1.8.2       AMXX Dev Team     imessage.amxx    running  

 [ 15] Admin Votes             1.8.2       AMXX Dev Team     adminvote.amxx   running  
 [ 16] NextMap                 1.8.2       AMXX Dev Team     nextmap.amxx     running  
 [ 17] Nextmap Chooser         1.8.2 f1    AMXX Dev Team     mapchooser.amxx  running  
 [ 18] TimeLeft                1.8.2       AMXX Dev Team     timeleft.amxx    running  
 [ 19] Pause Plugins           1.8.2       AMXX Dev Team     pausecfg.amxx    running  
 [ 20] Stats Configuration     1.8.2       AMXX Dev Team     statscfg.amxx    running  
 [ 21] StatsX                  1.8.2       AMXX Dev Team     statsx.amxx      running  
 [ 22] CS Misc. Stats          1.8.2       AMXX Dev Team     miscstats.amxx   running  
 [ 23] Quake Sounds            2.1         hleV              QuakeSounds.amx  running  
 [ 24] Ultimate Sounds         1.7         Dizzy / Bmann_42  ultimate_sounds  running  
 [ 25] Last man bets           0.9.4       JGHG              lastmanbets.amx  running  
 [ 26] Automatic Unstuck       1.5         NL)Ramon(NL       stuck.amxx       running  
 [ 27] Lame Connection Punish  1.2         MeRcyLeZZ         lame_connection  running  
 [ 28] AMXX Spy++              1.0         [DeathTV] Sid 6.  amx_spyplusplus  running  
 [ 29] Automatic knife duel    0.3         JGHG              automatic_knife  running  

 [ 30] Autoresponder/Advertis  0.5         MaximusBrood      ad_manager.amxx  running  
 [ 31] Ultimate Gore           1.6         JTP10181          amx_gore_ultima  running  
 [ 32] Pump Knockback (when s  1.0         v3x & Chronic     pump_knockback.  running  
 [ 33] AMX Super Cmd Manager   5.0.2       SuperCentral.co   amx_super-cmdma  running  
 [ 34] Amx Super Fun           5.0.2       SuperCentral.co   amx_super-fun.a  running  
 [ 35] AMX_Super Menu          5.0.2       SuperCentral.co   amx_super-menu.  running  
 [ 36] Amx Super Others - TES  5.0.2       SuperCentral.co   amx_super-other  running  
 [ 37] Amx Super Serious       5.0.2       SuperCentral.co   amx_super-serio  running  
( 11) Load fails: Plugin file open error (plugin "antiflood.amxx")
37 plugins, 36 running

Protocol version 48
Exe version 1.0.0.3/Stdio (czero)
Exe build: 19:53:27 Aug  3 2020 (8684)

hostname:  Botsylvania
version :  48/1.0.0.3/Stdio 8684 secure  (80)
tcp/ip  :  108.61.125.101:27015
map     :  cs_italy_cz at: 0 x, 0 y, 0 z
players :  13 active (18 max)

P.S. I really appreciate your help on this.

**Comment by xLeviNx** [comment]
[amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11343111/amx_super-others.zip)

**Comment by Fiercetree** [comment]
> [amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11343111/amx_super-others.zip)

Still the same issue with the join message:
Beware! (with a back slash)
Wolf has joined.

Instead of 
Beware!
Wolf has joined. 

And the leave message is not functioning.

I just wanted to know if I'm doing everything correctly. I'm compiling the .sma file and replacing the .amxx file and .sma file in the plugin and scripts folders, respectively. The plugin is already enabled in the third party plugin list in my plugin.ini folder, so this does not need to be changed. The plugin is running.   

I did test it with players and the leave message did not work, but I was also curious if simply kicking a bot from the server would trigger the leave message. 

Should I reinstall the complete package? 

[amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11348224/amx_super-others.zip)

This attached file is the compiled .amxx file from your link. Not sure if you wanted to check it out.

**Comment by xLeviNx** [comment]
> > [amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11343111/amx_super-others.zip)
> 
> Still the same issue with the join message: Beware! (with a back slash) Wolf has joined.
> 
> Instead of Beware! Wolf has joined.
> 
> And the leave message is not functioning.
> 
> I just wanted to know if I'm doing everything correctly. I'm compiling the .sma file and replacing the .amxx file and .sma file in the plugin and scripts folders, respectively. The plugin is already enabled in the third party plugin list in my plugin.ini folder, so this does not need to be changed. The plugin is running.
> 
> I did test it with players and the leave message did not work, but I was also curious if simply kicking a bot from the server would trigger the leave message.
> 
> Should I reinstall the complete package?
> 
> [amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11348224/amx_super-others.zip)
> 
> This attached file is the compiled .amxx file from your link. Not sure if you wanted to check it out.

Bots should not trigger the message, it should be players that do
As for the enter message, just remove the / in the line

I'm not sure what else to do. I've run this on a few 1.6 servers and tested in a cs:cz server and its functioning but periodically it does not

**Comment by Fiercetree** [comment]
> > > [amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11343111/amx_super-others.zip)
> > 
> > 
> > Still the same issue with the join message: Beware! (with a back slash) Wolf has joined.
> > Instead of Beware! Wolf has joined.
> > And the leave message is not functioning.
> > I just wanted to know if I'm doing everything correctly. I'm compiling the .sma file and replacing the .amxx file and .sma file in the plugin and scripts folders, respectively. The plugin is already enabled in the third party plugin list in my plugin.ini folder, so this does not need to be changed. The plugin is running.
> > I did test it with players and the leave message did not work, but I was also curious if simply kicking a bot from the server would trigger the leave message.
> > Should I reinstall the complete package?
> > [amx_super-others.zip](https://github.com/SuperCentral-Gaming-Community/AMX-Super-5.0.2/files/11348224/amx_super-others.zip)
> > This attached file is the compiled .amxx file from your link. Not sure if you wanted to check it out.
> 
> Bots should not trigger the message, it should be players that do As for the enter message, just remove the / in the line
> 
> I'm not sure what else to do. I've run this on a few 1.6 servers and tested in a cs:cz server and its functioning but periodically it does not

It's cool. I really appreciate you going the extra mile in trying to fix it. 

I can also understand when one creates something it should always work. But this problem is not a biggie to me. 

I can close this issue, or you can leave it open for now, in case anything changes or the next version may have a fix. (I get an email notifying me of any new messages on this thread). I also may update Amx mod to 1.9 in the future and test it. 

Thank you so much for your time and effort in this matter.

**Comment by Fiercetree** [comment]
_Update:_ I have upgraded to AMX Mod X 1.9 and the leave message is now working. I tried creating a custom leave message in the amxsuper.config file, but it still just puts the default in there ("thanks, come back soon"). This is fine though. I also wanted to point out when bots leave, the message does come up, but their bot name is not inculded with the message. Just wanted to let you know. 
_Further Update:_ After editing the custom leave message in cvar, the message works and adds the name of players. Still, when a bot leaves, their name is not listed and just shows the custom leave message. I feel from updating to AMX Mod X 1.9 the issue has been resolved. Thanks for your past help on this. 

In the future, I would probably just recommend users use 1.9 version or higher. Upgrading seems to have resolve alot of my issues with other things. Thanks for you help in this matter.
